src/hooks/useInterval.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from 'react';
+
+export default function useInterval(callback, delay, leading = true) {
+  const savedCallback = useRef();
+
+  // Remember the latest callback.
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  // Set up the interval.
+  useEffect(() => {
+    function tick() {
+      const { current } = savedCallback;
+      if (current) {
+        current();
+      }
+    }
+
+    if (delay !== null) {
```

Comment:
No need to run the interval if `savedCallback.current` is `undefined` (considering `if (current)` check above)